### DISCUSSION TITLE
Limit SQS Concurrency and/or control thread exhaustion

### DISCUSSION THREAD

**Discussion by shresthaujjwal** [awspring/spring-cloud-aws#469] [Q&A] [answered]
Hi Team, i understand there is major rewrite happening in v3 for SQS #344. But i am seeking an interim solution for an issue i am seeing in my environment. Here is my setup

- I have thousands of messages queue'd up in my 2 standard sqs. Around 20 messages/second
- 3 containers each running at 2Gb memory on m5.xlarge (4 CPU + 16GB Memory). All 3 containers is running on same instance
- simple microservice using spring cloud aws using 2 `SqsListener` 

Microservice runs for few min and eventually it will run out of thread and memory. What is the best way to handle load without exhausting the thread, please suggest the best practice. 

Error i get after few min of processing 100s of message
```
[1952.853s][warning][os,thread] Failed to start thread - pthread_create failed (EAGAIN) for attributes: stacksize: 1024k, guardsize: 0k, detached.
[1952.937s][warning][os,thread] Failed to start thread - pthread_create failed (EAGAIN) for attributes: stacksize: 1024k, guardsize: 0k, detached.
```

**Comment by maciejwalkowiak** [comment]
Hi. It is interesting as we have never had such issue reported so far. Share please your exact SQS integration configuration and used Java version

**Reply by shresthaujjwal** [reply]
thanks @maciejwalkowiak i have reported this as an issue. Please review #470

**Comment by shresthaujjwal** [comment, marked as answer]
closing this discussion, as it turns out my application had a memory leak. Apologize for inconvenience
